public/components/metrics/top_menu/__tests__/metrics_export.test.tsx

```diff
@@ -0,0 +1,188 @@
+      reducer: { metrics: metricsReducers, customPanel: panelReducer },
+      preloadedState,
+    }),
+    ...renderOptions
+  }: ExtendedRenderOptions = {}
+) {
+  function Wrapper({ children }: PropsWithChildren<{}>): JSX.Element {
+    return <Provider store={store}>{children}</Provider>;
+  }
+
+  // Return an object with the store and all of RTL's query functions
+  return { store, ...render(<Provider store={store}>{ui}</Provider>, { ...renderOptions }) };
+}
+
+describe('Export Metrics Panel Component', () => {
+  configure({ adapter: new Adapter() });
+  const store = createStore(rootReducer, applyMiddleware(thunk));
+  coreRefs.savedObjectsClient.find = jest.fn(() =>
+    Promise.resolve({
+      savedObjects: [],
+      then: () => Promise.resolve(),
+    })
+  );
+
+  it('renders Export Metrics Panel Component', async () => {
+    mockSavedObjectActions({ get: [{ savedVisualization: sampleSavedMetric }] });
+
+    const wrapper = mount(
+      <Provider store={store}>
+        <MetricsExport />
+      </Provider>
+    );
+    wrapper.update();
+
+    await waitFor(() => {
+      expect(wrapper).toMatchSnapshot();
+    });
+  });
+  it('opens metric export panel on Save button, closes on Cancel', async () => {
+    // Arrange
+    const preloadedState = {
+      metrics: {
+        ...defaultInitialState,
+        metrics: { [sampleSavedMetric.id]: sampleSavedMetric },
+        selectedIds: [sampleSavedMetric.id],
+      },
+      customPanel: {
+        panelList: [],
+      },
+    };
+
+    // Act
+    renderWithMetricsProviders(<MetricsExport />, { preloadedState });
+
+    // Assert
+    const saveButton = await screen.findByText(/Save/);
+    expect(saveButton).toBeInTheDocument();
+    fireEvent.click(saveButton);
+
+    const modalCancelButton = await screen.findByTestId('metrics__SaveCancel');
+    expect(modalCancelButton).toBeInTheDocument();
+    fireEvent.click(modalCancelButton);
+
+    await waitFor(() => {
+      expect(screen.queryByTestId('metrics__SaveCancel')).not.toBeInTheDocument();
+    });
+
+    expect(screen.queryByTestId('metrics__SaveCancel')).toBeNull();
+  });
+
+  it('opens metric export panel on Save button, closes by clicking away from panel', async () => {
+    // Arrange
+    const preloadedState = {
+      metrics: {
+        ...defaultInitialState,
+        metrics: { [sampleSavedMetric.id]: sampleSavedMetric },
+        selectedIds: [sampleSavedMetric.id],
+      },
+      customPanel: {
+        panelList: [],
+      },
+    };
+
+    // Act
+    renderWithMetricsProviders(
+      <div>
+        <span>A Clickable Target</span>
+        <MetricsExport />
+      </div>,
+      { preloadedState }
+    );
+
+    // Assert
+    const saveButton = await screen.findByText(/Save/);
+    expect(saveButton).toBeInTheDocument();
+    fireEvent.click(saveButton);
+
+    expect(await screen.findByText('Custom operational dashboards/application'));
+
+    userEvent.keyboard('{ESCAPE}');
```

Comment:
Can replace this with `fireEvent.keyDown(element, { key: 'Escape', code: 'Escape' });`  As we don't have `'@testing-library/user-event'` installed.

Reply:
Done.